src/customJs/properties/promo_codes/PromoCodesWidget.test.js

```diff
@@ -183,4 +177,20 @@ describe('Promocode widget', () => {
     const spinnerContainer = await screen.findByRole('status');
     expect(spinnerContainer).toBeDefined();
   });
+
+  it('must be render no promo codes available message', async () => {
+    requestDopplerApp.mockImplementation((params) => {
+      params.callback([]);
+      const destructor = () => {};
+      return destructor;
```

Comment:
If I am not wrong, it should be in this way:

```suggestion
      return { destructor };
```

But, if the test pass, probably, none of this code is necessary.